openmmtools/tests/test_integrators.py

```diff
@@ -312,3 +313,117 @@ def test_thermostated_integrator_hash():
+                                                                  splitting="O { V R H R V } O",
+                                                                  nsteps_neq=nsteps,
+                                                                  direction="reverse")
+
+    platform = openmm.Platform.getPlatformByName("Reference")
+
+    #Do 100 iterations of each direction
+    n_iterations = 10000
+
+    #instantiate the testsystem
+    alanine_dipeptide = testsystems.AlanineDipeptideVacuum()
+
+    #alchemically modify everything:
+    n_atoms = alanine_dipeptide.system.getNumParticles()
+    alchemical_factory = alchemy.AlchemicalFactory(consistent_exceptions=False)
+    alchemical_region = alchemy.AlchemicalRegion(range(n_atoms))
+    modified_system = alchemical_factory.create_alchemical_system(reference_system=alanine_dipeptide.system,
+                                                                  alchemical_regions=alchemical_region)
+
+    alchemical_ctx_forward = openmm.Context(modified_system, alchemical_integrator_forward, platform)
+    alchemical_ctx_reverse = openmm.Context(modified_system, alchemical_integrator_reverse, platform)
+
+    #get the forward work values:
+    positions = alanine_dipeptide.positions
+    w_f = numpy.zeros([n_iterations])
+    for i in range(n_iterations):
+        w_f[i], eq_positions = run_nonequilibrium_switching(modified_system, positions,
+                                              default_functions, alchemical_integrator_forward, 100, alchemical_ctx_forward,
+                                              direction="forward")
+        positions = eq_positions
+
+        print(i)
+
+    #get the reverse work values:
+    w_r = numpy.zeros([n_iterations])
+    for i in range(n_iterations):
+        w_r[i], eq_positions = run_nonequilibrium_switching(modified_system, positions,
+                                              default_functions, alchemical_integrator_reverse, 100, alchemical_ctx_reverse,
+                                              direction="reverse")
+        positions = eq_positions
+        print(i)
+
+    deltaF, ddeltaF = pymbar.BAR(w_f, w_r)
+
+    print(deltaF)
+    print(ddeltaF)
+
+
+
+def run_nonequilibrium_switching(system, positions, alchemical_functions, alchemical_integrator, nsteps, alchemical_ctx, direction="forward"):
+    """
+    Equilibrate and then run some nonequilibrium switching simulations
+
+    Parameters
+    ----------
+    system
+    positions
+    alchemical_integrator
+    direction
+
+    Returns
+    -------
+    protocol_work : float
+        Work performed by protocol
+    """
+
+    #make a ghmc integrator with the default parameters
+    ghmc_integrator = GHMCIntegrator()
+
+    #use the reference platform, since this is for a test
+    platform = openmm.Platform.getPlatformByName("Reference")
+
+    #make a context
+    context = openmm.Context(system, ghmc_integrator, platform)
```

Comment:
Can probably move this out of the inner-loop function -- the test creates an OpenMM context 20,000 times, when the only thing that's changing between iterations is the initial positions, I think